mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/gateway/tasks/BasicStructureMigrationTask.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020, Mulesoft, LLC. All rights reserved.
+ * Use of this source code is governed by a BSD 3-Clause License
+ * license that can be found in the LICENSE.txt file.
+ */
+package com.mulesoft.tools.migration.library.gateway.tasks;
+
+import static com.mulesoft.tools.migration.project.ProjectType.MULE_FOUR_POLICY;
+import static com.mulesoft.tools.migration.util.MuleVersion.MULE_3_VERSION;
+import static com.mulesoft.tools.migration.util.MuleVersion.MULE_4_VERSION;
+import static java.util.Collections.singleton;
+
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.AfterExceptionTagMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.AfterTagMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.BeforeExceptionTagMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.BeforeTagMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.CleanupAttributesMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.DataTagMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.PointcutTagMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.PolicyFileRenameMigrationStep;
+import com.mulesoft.tools.migration.library.gateway.steps.policy.basicstructure.PolicyTagMigrationStep;
+import com.mulesoft.tools.migration.project.ProjectType;
+import com.mulesoft.tools.migration.step.MigrationStep;
+import com.mulesoft.tools.migration.task.AbstractMigrationTask;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Basic policy structure migration
+ *
+ * @author Mulesoft Inc.
+ */
+public class BasicStructureMigrationTask extends AbstractMigrationTask {
+
+  @Override
+  public String getDescription() {
+    return "Basic Policy structure migration";
+  }
+
+  @Override
+  public String getTo() {
+    return MULE_4_VERSION;
+  }
+
+  @Override
+  public String getFrom() {
+    return MULE_3_VERSION;
+  }
+
+  @Override
+  public Set<ProjectType> getApplicableProjectTypes() {
+    return singleton(MULE_FOUR_POLICY);
```

Comment:
@Pgarrett this is the only task that specifies that applies to policy projects only. Should we add this to the rest of the tasks? Because if this is not defined, they will run on any project (application, domain and policy).

Reply:
You're right! We should add this to the following tasks:

- ClientIdEnforcementMigrationTask
- FederationMigrationTask
- HttpMigrationTask
- IpFilterMigrationTask
- MuleElementsMigrationTask
- ThreatProtectionMigrationTask
- ThrottlingMigrationTask

Reply:
Ok, I will add it there, thanks!